### PR TITLE
`Trusted Entitlements`: produce verification failures for static endpoints with no signature

### PR DESCRIPTION
**Pull request by NachoSoto**
Equivalent to https://github.com/RevenueCat/purchases-android/pull/1119

### CODE REVIEW

File: Sources/Security/Signing+ResponseVerification.swift [hunk 1]
```diff
@@ -95,3 +94,16 @@ extension HTTPResponse where Body == Data? {
...
+
+private extension HTTPRequest {
+
+    var requiresSignature: Bool {
+        guard self.path.supportsSignatureVerification else {
+            return false
+        }
+
+        let needsNonce = self.path.needsNonceForSigning
+        return !needsNonce || (needsNonce && self.nonce != nil)
```
Comment (tonidero) [inline Sources/Security/Signing+ResponseVerification.swift:106]: Hmm I wonder, is this part needed? In Android I'm only checking that `supportsSignatureVerification` is true and that entitlement verification is enabled. In that situation, we should always require a signature I think? Whether the path requires a nonce or not, shouldn't affect whether we require a signature I think.

File: Sources/Security/Signing+ResponseVerification.swift [hunk 2]
```diff
@@ -95,3 +94,16 @@ extension HTTPResponse where Body == Data? {
     }
 
 }
+
+private extension HTTPRequest {
+
+    var requiresSignature: Bool {
```
Comment (tonidero) [inline Sources/Security/Signing+ResponseVerification.swift:100]: Should this also check if entitlement verification is enabled?

File: Sources/Security/Signing+ResponseVerification.swift (hunk 1 above)
Comment (NachoSoto) [reply Sources/Security/Signing+ResponseVerification.swift:106]: Oh the issue was some of the failing tests were wrong. I thought I needed this check but I don't! Fixed.

File: Sources/Security/Signing+ResponseVerification.swift (hunk 2 above)
Comment (NachoSoto) [reply Sources/Security/Signing+ResponseVerification.swift]: That's done by the calling method. But I removed this property so it's more clear now.

**Pull request merged**